solutions/src/main/java/ru/sberbank/school/task13/cacheproxy/cache/CacheProxyImpl.java

```diff
@@ -0,0 +1,99 @@
+package ru.sberbank.school.task13.cacheproxy.cache;
+
+import lombok.NonNull;
+import ru.sberbank.school.task13.cacheproxy.annotation.Cache;
+import ru.sberbank.school.task13.cacheproxy.exception.CacheProxyException;
+import ru.sberbank.school.task13.cacheproxy.serialization.SerializationManager;
+import ru.sberbank.school.task13.cacheproxy.utils.CacheProxyUtils;
+
+import java.io.*;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+
+public class CacheProxyImpl implements InvocationHandler {
+    private Object delegate;
+    private Map<String, Object> cache;
+    private final String fileDirectory;
+    private final SerializationManager serializationManager;
+
+    public CacheProxyImpl(@NonNull Object delegate, String fileDirectory) {
+        this.delegate = delegate;
+        this.cache = new HashMap<>();
+        this.fileDirectory = fileDirectory;
+        this.serializationManager = new SerializationManager();
+    }
+
+    @Override
+    public Object invoke(Object proxy, @NonNull Method method, Object[] args)
+            throws CacheProxyException {
+        Object result = null;
+
+        if (!method.isAnnotationPresent(Cache.class)) {
+            result = methodInvoke(method, delegate, args);
+        } else {
+            Cache annotations = method.getDeclaredAnnotation(Cache.class);
+            String key = CacheProxyUtils.getKey(annotations.identityBy(),
+                    annotations.key(),
+                    method.getName(),
+                    args);
+            switch (annotations.getCacheType()) {
```

Comment:
switch without &quot;default&quot; clause.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1859708796>*